src/main/java/com/google/sps/data/UserQuery.java

```diff
@@ -14,47 +14,128 @@
 
 package com.google.sps.data;
 
-import com.google.sps.util.SearchUtils;
+import com.google.appengine.api.datastore.Query;
+import java.lang.UnsupportedOperationException;
 import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
 
 /**
  * Represents the separated user query data which can be used to run different Datastore
  * query requests.
  */
 public class UserQuery {
+  /**
+   * TODO: Implement the tag and author collections as sets to avoid duplicates
+   * NOTE: The methods used are from collection interface, so a change in implementation won't break
+   * Existing code
+   */
   private ArrayList<String> tagsAndIngredients;
   private ArrayList<String> authors;
   private String recipeName;
 
   /**
-   * This constructor takes in a raw user query string, with special query symbols, and initializes
-   * instance variables using methods in SearchUtils to handle the query.
-   * TODO: Implement regex and make calls to initialize authors, and tagsAndIngredients using regex functions
+   * The constructor takes in the string parameters for the query from the search bar, as well
+   * as strings for the tags and authors. These will be passed from the front end with delimeters,
+   * so these instance variables are initialized using helper functions which will return the tags
+   * in an ArrayList
+   * NOTE: the helper functions will initialize tagsAndIngredients and authors to be empty collections
+   * if the param value results in empty strings or is null.
+   * TODO: implement initialization of tags and authors
    */
-  public UserQuery(String query) {
-    String standardizedQuery = SearchUtils.standardize(query);
+  public UserQuery(String query, String tags, String authors) {
+    this.recipeName = query;
     this.tagsAndIngredients = new ArrayList<String>();
     this.authors = new ArrayList<String>();
-    this.recipeName = SearchUtils.parseRecipeName(standardizedQuery);
   }
 
   /**
-   * Takes a new tag or ingredient to search for and adds it to the tagsAndIngredients attribute.
-   * TODO: Implement ability to avoid duplicates
+   * Returns a filter which can be used to construct a Datastore query based on the instance variables
+   * of this object. If the object has no data which can be used to construct filters, it will return
+   * a single filter for making sure recipes are public.
    */
-  public void addTag(String tag) {
-    this.tagsAndIngredients.add(tag);
+  public Query.Filter createSearchFilter() {
+    // ArrayList which will hold filters for name, authors, tags, and visibility.
+    ArrayList<Query.Filter> searchFilters = new ArrayList<Query.Filter>();
+    
+    // Creates and adds filters to the list if there is data to make the filter
+    if (tagsAndIngredients.size() > 0){
+      searchFilters.add(createTagsFilter());
+    }
+    if (authors.size() > 0) {
+      searchFilters.add(createAuthorsFilter());
+    }
```

Comment:
Use `!authors.isEmpty()`, same for tagsAndIngredients. In general, it's more readable

Reply:
Ack and fixed.

Reply:
I had isEmpty(), but then when making my code check for the `size() == 1` case, I switched the if statement to be `size() > 1` because I though it was more consistent. Lmk if you don't agree and I'll make the change to `isEmpty()`